gui

```diff
@@ -1,28 +1,285 @@
+      fi
+    fi
+    
+    num_users="$(usercount "$app")"
+    if [ ! -z "$num_users" ] && [ "$num_users" -gt 20 ];then
+      #list the number of users, using this printf command to add commas (,) for every thousand number
+      abovetext+=$'\n'"- <b>$(printf "%'d" "$num_users")</b> users"
+      
+      if [ "$num_users" -ge 10000 ];then
+        #if this app has over 10,000 users, add two exclamation points!!
+        abovetext+="!!"
+      elif [ "$num_users" -ge 1500 ];then
+        #if this app has over 1500 users, add an exclamation point!
+        abovetext+="!"
+      fi
+    fi
+    
+    #array holding various buttons that may be passed to yad
+    whichbutton=()
+    
+    if [ -f "${DIRECTORY}/apps/$app/uninstall" ] || [ -f "${DIRECTORY}/apps/$app/install" ] || [ -f "${DIRECTORY}/apps/$app/install-$arch" ];then
+      whichbutton+=("--button=Scripts!${DIRECTORY}/icons/shellscript.png!Feel free to see how an app is installed!"$'\n'"Perfect for learning or troubleshooting.:6")
+    fi
+    if [ "$(cat "${DIRECTORY}/data/settings/Show Edit button")" == 'Yes' ];then
+      #if edit button enabled, show it
+      whichbutton+=("--button=Edit!${DIRECTORY}/icons/edit.png!Make changes to the app:8")
+    fi
+    
+    #display buttons based on app's installation status
+    if [ "$app_status" == 'installed' ];then
+      #if installed, display uninstall button
+      whichbutton+=("--button=Uninstall!${DIRECTORY}/icons/uninstall.png:2")
+    elif [ "$app_status" == 'uninstalled' ];then
+      #if uninstalled, display install button
+      whichbutton+=("--button=Install!${DIRECTORY}/icons/install.png:4")
+    elif [ "$app_status" == 'disabled' ];then
+      #if disabled, display only an 'enable' button
+      whichbutton+=("--button=<b>Enable</b>!!Force this app to install on your system."$'\n'"This app was disabled for a reason so if you enable it..."$'\n'"YOU HAVE BEEN WARNED.:12")
+    else
+      #if app status is 'corrupted', and a log file exists for this app, then display a button to view the log file
+      if [ "$app_status" == 'corrupted' ];then
+        logfile="$(ls "$DIRECTORY/logs"/* -t | grep "fail-${app}" | head -n1)"
+        if [ ! -z "$logfile" ];then
+          whichbutton+=("--button=Errors!${DIRECTORY}/icons/log-file.png!$app failed to $(echo "$(basename "$logfile")" | awk -F'-' '{print $1}'). Click this button to view the error output saved in the log file.:14")
+        fi
+      fi
+      #if status is corrupted or unknown, then show both buttons
+      whichbutton+=("--button=Uninstall!${DIRECTORY}/icons/uninstall.png:2" "--button=Install!${DIRECTORY}/icons/install.png:4")
+    fi
+    
+    #only display app icon if it exists
+    if [ -f "${DIRECTORY}/apps/${app}/icon-64.png" ];then
+      imageline=(--image="${DIRECTORY}/apps/${app}/icon-64.png" --image-on-top)
+    fi
+    
+    (cat "${DIRECTORY}/apps/$app/description" || echo "Description unavailable") | yad --text-info --fontname=12 --wrap --show-uri \
```

Comment:
```suggestion
    (while [ "$(xdotool getactivewindow getwindowname)" != "Details of ${app}" ]; do wmctrl -a "Details of ${app}"; done) &
    (cat "${DIRECTORY}/apps/$app/description" || echo "Description unavailable") | yad --text-info --fontname=12 --wrap --show-uri \
```

Automatically focus on app info window.

Reply:
> Automatically focus on app info window.

```bash
(while [ "$(xdotool getactivewindow getwindowname)" != "Details of ${app}" ]; do wmctrl -a "Details of ${app}"; done) &
```
That seems to be a bit dangerous without a delay in the loop.

Reply:
@cycool29 I've added your suggestion with a few safety improvements.